Lista2/Ex1/ex1-Part1/main.c

```diff
@@ -0,0 +1,138 @@
+#include <stdlib.h>
+#include <stdio.h>
+#define STACKSIZE 10
+
+
+
+typedef struct stack{
+    int top;
+    int stackItems[STACKSIZE];
+} Stack;
+
+
+void startStack(Stack *stack){
+    stack->top = -1; // inicializando a pilha
+}
+
+int empty(Stack *stack){
+    return (stack->top == -1) ? 1:0;
+}
+
+void push(Stack *stack, int number){
+    if (stack->top + 1 == STACKSIZE){
+        printf("\nOverflow\n");
+    }
+    else{
+        stack->top += 1; // anda com o topo e depois adiciona
+        stack->stackItems[stack->top] = number;
+        printf("\nElemento adicionado");
+    }
+}
+
+void pop(Stack *stack){
+    if(empty(stack)){
+        printf("\nUnderflow\n ");
+    }
+    else{
+        int returnedValue = stack->stackItems[stack->top];
+        printf("\n Element removed: %d\n", returnedValue);
+        stack->stackItems[stack->top] = 0;
+        stack->top--;
```

Comment:
Aqui não seria mais interessante colocar NULL ao invés de 0? Porque teoricamente o 0 ainda pode representar um int

Reply:
sim, sim, mas a minha idéia é só apagar o valor anterior. E como to lidando com vetor, não sei se vai dar bom colocar NULL ali. Mas posso testar